lib/puppet/file_serving/metadata.rb

```diff
@@ -22,10 +22,16 @@ def checksum_type=(type)
     @checksum_type = type
   end
 
+  def source_permissions=(source_permissions)
+    raise(ArgumentError, "Unsupported source_permission #{source_permissions}") unless [:use, :use_when_creating, :ignore].include?(source_permissions.intern)
```

Comment:
I would expect a `puppet agent` run to serialize/deserialize the `source_permissions` option as a String, and then call the setter with a string. It may be that `puppet apply` or running via tests is calling with a symbol? I'm thinking we should be consistent about what we store in the instance variable, so that the comparison below `source_permissions == :ignore` work as expected.